docs/examples.md

```diff
@@ -312,3 +312,111 @@ steps:
           push:
             - app:myregistry:port/myrepo/myapp:latest
 ```
+
+### Create, Use and Remove Builder Instances
+
+#### Create
+
+Most Docker setups, unless configured, will use the `docker` Builder Driver by default. More details on it [here](https://docs.docker.com/build/builders/drivers/docker/).
+
+The `docker` driver can handle most situations but for advance features with the Docker, different Builder Drivers are required and this requires a Builder Instance being created first, which can be done manually or with the Plugin. To create a Builder Instance using a chosen Driver, requires the `name`, `driver` and `create` parameters:
+
+```yaml
+steps:
+  - label: ":docker: Build an image"
+    plugins:
+      - docker-compose#v5.4.1:
+          build: app
+          push: app:index.docker.io/myorg/myrepo:my-branch
+          cache-from:
+            - "app:myregistry:port/myrepo/myapp:my-branch"
+            - "app:myregistry:port/myrepo/myapp:latest"
+          builder:
+            name: container
+            driver: docker-container
+            create: true
+```
+
+**If a Builder Instance with the same `name` already exists, it will not be recreated.**
+
+#### Use
+
+By default, Builder Instances specified by `name` or that are created with `create` are not used, and the default Builder Instance on the Agent will be used. To use a Builder Instance, requires the `name` and `use` parameters and the Builder Instance to exist:
+
+```yaml
+steps:
+  - label: ":docker: Build an image"
+    plugins:
+      - docker-compose#v5.4.1:
+          build: app
+          push: app:index.docker.io/myorg/myrepo:my-branch
+          cache-from:
+            - "app:myregistry:port/myrepo/myapp:my-branch"
+            - "app:myregistry:port/myrepo/myapp:latest"
+          builder:
+            name: container
+            use: true
+```
+
+#### Remove
+
+By default, Builder Instances specified by `name` or that are created with `create` are not removed after the Job finishs. To remove a Builder Instance, requires the `name` and `remove` parameters and the Builder Instance to exist:
+
+```yaml
+steps:
+  - label: ":docker: Build an image"
+    plugins:
+      - docker-compose#v5.4.1:
+          build: app
+          push: app:index.docker.io/myorg/myrepo:my-branch
+          cache-from:
+            - "app:myregistry:port/myrepo/myapp:my-branch"
+            - "app:myregistry:port/myrepo/myapp:latest"
+          builder:
+            name: container
+            remove: true
+```
```

Comment:
Based on the previous comment, this example would not actually create or use the builder defined :S